zookeeper-server/src/test/java/org/apache/zookeeper/server/ReferenceCountedACLCacheTest.java

```diff
@@ -284,4 +319,29 @@ private Long convertACLsNTimes(ReferenceCountedACLCache cache, List<ACL> acl, in
         return acl1;
     }
 
+    private void serializeAndDeserializeACL(ReferenceCountedACLCache cache,
+                                            ReferenceCountedACLCache deserializedCache) throws IOException {
+        if (System.getProperty(SnapshotFactory.ZOOKEEPER_SNAPSHOT_NAME).equals(
+                FileSnap.class.getName())) {
+            LOG.info("Testing FileSnap:");
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            BinaryOutputArchive archive = BinaryOutputArchive.getArchive(baos);
+            FileSnap snapLog = new FileSnap(null, archive);
+            snapLog.serializeACL(cache);
+            BinaryInputArchive inArchive = BinaryInputArchive.getArchive(new ByteArrayInputStream(baos.toByteArray()));
+            snapLog = new FileSnap(null, inArchive);
+            snapLog.deserializeACL(deserializedCache);
+            snapLog.close();
+        } else {
+            LOG.info("Testing RocksDBSnap:");
```

Comment:
Please test for the expected value, in the future we will have new implementations

Reply:
Good suggestion, updated with explicit alternative conditions.